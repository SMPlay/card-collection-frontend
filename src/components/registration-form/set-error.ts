type ValidateError = boolean | string | undefined;

export const setError = (status: string, validateError: ValidateError, serverError: string) => {
  if (status === "error") {
    return serverError;
  }

  return validateError;
}
