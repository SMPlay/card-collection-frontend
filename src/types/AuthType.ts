export interface AuthType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  handleBlur: (e: any) => void;
  values: T;
  status: string;
  errors: T;
  touched: T;
}
