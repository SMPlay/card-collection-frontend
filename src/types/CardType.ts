export interface CardType {
  id: string;
  name: string;
  imageUrl: string;
  role: string;
  rarity: string;
  need: number;
  have: number;
  number: number;
}