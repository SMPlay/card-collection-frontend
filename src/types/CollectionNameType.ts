export interface CollectionNameType {
  id: string;
  name: string;
}
