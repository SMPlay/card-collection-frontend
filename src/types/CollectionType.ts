export interface CollectionType {
  name: string;
  imageUrl: string;
  id: string;
  cardsCount: number;
  release: number;
}
