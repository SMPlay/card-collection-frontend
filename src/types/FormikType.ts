interface Status {
  loading: boolean;
  error: string;
}

export interface FormikType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  values: T;
  status: Status;
  errors: T;
}
