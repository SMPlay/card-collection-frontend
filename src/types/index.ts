export { CollectionType } from './CollectionType'
export { PageType } from './PageType'