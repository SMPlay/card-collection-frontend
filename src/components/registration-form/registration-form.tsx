import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import { Box, Button, TextField, Typography } from "@material-ui/core";

import { AuthType } from "../../types/AuthType";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      "& .MuiTextField-root": {
        margin: theme.spacing(1),
        width: "60ch",
      },
    },
    registrationField: {
      display: "flex",
      flexDirection: "column",
    },
    buttonsField: {
      display: "flex",
      flexDirection: "row",
      alignItems: "center",
    },
  })
);

export interface Values {
  login?: string | boolean;
  password?: string | boolean;
  confirmPassword?: string | boolean;
}

export const RegistrationForm: React.FC<AuthType<Values>> = ({
  handleSubmit,
  handleChange,
  handleBlur,
  values,
  status,
  errors,
  touched,
}) => {
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit} className={styles.root}>
      {status?.error ? (
        <Typography variant="h4" component="h2">
          {status.error}
        </Typography>
      ) : (
        ""
      )}
      <Box m={2} className={styles.registrationField}>
        <TextField
          id="login"
          value={values.login}
          onChange={handleChange}
          onBlur={handleBlur}
          label="Логин"
          placeholder="Введите логин"
          variant="outlined"
          error={
            errors.login ||
            status?.error === "Такой пользователь уже существует"
              ? true
              : false
          }
          helperText={
            status?.error === "Такой пользователь уже существует"
              ? status?.error
              : ""
          }
          required
          disabled={status?.loading}
        />
        <TextField
          id="password"
          value={values.password}
          onChange={handleChange}
          onBlur={handleBlur}
          label="Пароль"
          placeholder="Введите пароль"
          variant="outlined"
          type="password"
          error={errors.password ? true : false}
          helperText={errors.password}
          required
          disabled={status?.loading}
        />
        <TextField
          id="confirmPassword"
          value={values.confirmPassword}
          onChange={handleChange}
          onBlur={handleBlur}
          label="Пароль ещё раз"
          placeholder="Введите пароль снова"
          variant="outlined"
          type="password"
          error={errors.confirmPassword ? true : false}
          helperText={errors.confirmPassword}
          required
          disabled={status?.loading}
        />
      </Box>
      <Button
        variant="contained"
        color="primary"
        type="submit"
        disabled={status?.loading}
      >
        Регистрация
      </Button>
    </form>
  );
};
