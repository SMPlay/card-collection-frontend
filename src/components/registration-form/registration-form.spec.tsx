import React from "react";
import { create } from "react-test-renderer";

import { RegistrationForm, Values } from "./registration-form";
import { AuthType } from "../../types/AuthType";

describe("Registration from component", () => {
  let props: AuthType<Values>;

  beforeEach(() => {
    props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      values: {
        login: "test",
        password: "test",
        confirmPassword: "test"
      },
      status: {
        loading: false,
        error: "",
      },
      errors: {
        login: "",
        password: "",
        confirmPassword: "test"
      }
    }
  });

  it("should render", () => {
    const tree = create(
      <RegistrationForm {...props}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
