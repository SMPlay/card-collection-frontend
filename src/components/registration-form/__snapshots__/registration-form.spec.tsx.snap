// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration from component should render 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[MockFunction]}
>
  
  <div
    className="MuiBox-root MuiBox-root-4 makeStyles-registrationField-2"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="login"
        id="login-label"
      >
        Логин
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="login"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите логин"
          required={true}
          type="text"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
          >
            <span>
              Логин
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        Email
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите email"
          required={true}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Email
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="password"
        id="password-label"
      >
        Пароль
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите пароль"
          required={true}
          type="password"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
          >
            <span>
              Пароль
               *
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="confirmPassword"
        id="confirmPassword-label"
      >
        Пароль ещё раз
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="confirmPassword-helper-text"
          aria-invalid={true}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="confirmPassword"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Введите пароль снова"
          required={true}
          type="password"
          value="test"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
          >
            <span>
              Пароль ещё раз
               *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled Mui-required"
        id="confirmPassword-helper-text"
      >
        test
      </p>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Регистрация
    </span>
  </button>
</form>
`;
